src/Func.h

```diff
@@ -59,6 +64,8 @@ class Func : public BroObj {
 	void Describe(ODesc* d) const override = 0;
 	virtual void DescribeDebug(ODesc* d, const val_list* args) const;
 
+	virtual Val* DoClone();
```

Comment:
It looks like you're implementing the API from `Val` here, which I think isn't really necessary, as `Func` is outside of that hierarchy. I'd change this to `virtual Func* Clone()`, and then override that in derived classes.